### ISSUE TITLE
Why can I not just download the file?

### ISSUE THREAD

**Issue by NessaishereDuck** [scriptish/scriptish#116]
This used to be the way it worked. Why the change?  Protecting me from myself?

If you are protecting me from myself, Please stop! Give me the option!!! Please!

**Comment by nmaier** [comment]
First of all: keep calm, please. Nobody accused you of anything.

What exactly are you talking about?

**Comment by LouCypher** [comment]
@nmaier it's related to #115

**Comment by nmaier** [comment]
Ok, so this is basically a dupe of #115. Closing

**Comment by NessaishereDuck** [comment]
OK did I say I was accused? Sry If I did but I don't see that. I am calm I just would like to be able to download the actual file without needing to do the source and then do a make. I would like the option of just having it install like the button does now or, downloading the script in such a manner as was always the norm and then when I opened in my addons for any browser it would then install. All I asked was that you not protect me from being stupid and needing the app to install the file, that fine that it does I would like the option though. 

I don't see where I am saying I was accused of anything, and the post is a little strong, but I don't understand your reaction. sry.

Did you ever use GM or Sciptish before it did the install it for you thing? In the past, whenever you wanted or needed a script, you would click on a button and download the file. I don't think that is difficult to do and it would allow non-supported browsers that can use Scriptish just fine, actually use it. Like Avant for one. Another big plus is if for any reason you need to uninstall the addon,, you do not need to redownload it. Just open it in your addons area and click install. Does that make any sense to you? Let me find you a GM script that still works the old way...

http://userscripts.org/scripts/show/92257

This does not auto install it either downloads a file or gives you the script needed to install the thing. From there I can save the code then open it in my add-ons section of my browser and it asks me if I want to install it. I click yes, and it then installs in that browser. This script is for a site called conquerclub and it manages your on site chat for you very nice to have. Does this make sense now?

**Comment by nmaier** [comment]
> Does that make any sense to you?

No, I cannot understand exactly what you're trying to tell us here. Please give actual steps to reproduce, like:

> - First do this
> - Then do that
> - Last click there
> 
> Expected:
> It should have done xy.
> 
> Actual:
> It did something else or gave an error message.

**Comment by NessaishereDuck** [comment]
OK Nils, I will try to be stepwise specific,

1] Use a browser that does not have GM or scriptish installed, that is the easiest way to grab the source code
2] go to http://userscripts.org/scripts/show/92257 http://userscripts.org/scripts/show/92257&lang=en  This is a file that has many users and is well trusted on the conquerclub site
3] Click on the green Install button in the upper right of the scrreen
4] save the file awith the appropriate file type that scriptish now uses - I can't install scriptish so cannot tell you for sure, but I would hope you would know.
5] open a browser that has scriptish or gm installed
6] Go to the addons area of that browser for firefox that would be: Tools menu, then click on Addons. Then select plugins
7] open the file you just saved on your harf frive. It will ask you if you want to install the addon

I hope that helps.

Nessa 
----- Original Message -----

**Comment by nmaier** [comment]
> 5] open a browser that has scriptish or gm installed

JFYI, we don't support Greasemonkey here.

> 7] open the file you just saved on your harf frive. It will ask you if you want to install the addon

How do you open it?

Alright, I understand the script, but you didn't really say:
- What did you expect to happen?
- What did actually happen?

**Comment by NessaishereDuck** [comment]
Hello Nils,

How to open the file - you open your addons area 
for example in firefox click on the tools menu then click AddOns
Then you simply drag the file from windows explorer or your desktop, where ever you have it and drop it on the addons page. 
At that point it will ask you if you want to install it.

What I hoped would happen is that I would be prompted to download the file instead of my only option being : Install to firefox. Let me rephrase. when I click the button it automatically tries to install scriptish on firefox. What is hoped instead is that I could download the file and then drag and drop it on my addons page.

**Comment by nmaier** [comment]
I don't get it. The file is already on your hard disk, and you drop it and you want to download it again?

**Comment by NessaishereDuck** [comment]
No once you have it on your hard drive you drag and drop it onto your addons page. This will install the script, not download it again.

If you are referring to my "what I hope would happen", I do not have scriptish on my hard drive. So I cannot drop it onto my addons page. So I need the ability to download the file so that I can install it the old way.

**Comment by nmaier** [comment]
> If you are referring to my "what I hope would happen", I do not have scriptish on my hard drive. So I cannot drop it onto my addons page. So I need the ability to download the file so that I can install it the old way.

Your are always talking about "download the file". What file? The user script is on your harddisk already you said.

**Comment by NessaishereDuck** [comment]
No my friend, I do not have the scriptish file because I cannot download it. I have other addons on my hard drive, a few hundred actually, that I install when I need them then uninstall when I am finished using them.

**Comment by nmaier** [comment]
Ah. You're actually just after the scriptish .xpi (installer)?
Well, you can download that from https://addons.mozilla.org/addon/scriptish
(Right click the Install button and save as).
Shortcut, always pointing to the latest version: https://addons.mozilla.org/firefox/downloads/latest/231203/addon-231203-latest.xpi
Old versions: https://addons.mozilla.org/addon/scriptish/versions

**Comment by NessaishereDuck** [comment]
I have firefox but it does not give me the control over every aspect of my browsing. I have seamonkey and I was a fan for a long time. Then they took away my ability to control every aspect of my browser, and I cannot live with that. Avant is my current browser. I am a power user Nils. I have addons for programming, games, and all sorts of app specific addons. One game Kingdom Of Loathing [or KoL] I have over 100 addons just for that one game. I have over 50 profiles installed so I can quickly switch to an entire new set of addons for a specific app, like KoL. So I really don't want to use more than one browser and I don't have enough memory to run more than one browser as I use a ton of memory for a dozen things I might be working on at any point in time. So it would be really awesome if I could somehow install scriptish to it.

**Comment by NessaishereDuck** [comment]
OK Thanks Nils I will give that a try.

**Comment by NessaishereDuck** [comment]
Well, you can download that from https://addons.mozilla.org/addon/scriptish 
When I get here, I see an Add to firefox button but if I right click I see no "save as "

what browser are you using and I will try with your browser

**Comment by NessaishereDuck** [comment]
OK never mind my friend. I used your next link 

Shortcut, always pointing to the latest version to the latest version: https://addons.mozilla.org/firefox/downloads/latest/231203/addon-231203-latest.xpi

And I was able to down load it from there. Thank you Nils.

**Comment by nmaier** [comment]
I'm using Firefox.
![saveas](https://f.cloud.github.com/assets/383665/751860/8dd581a2-e51d-11e2-9f9b-9550a53957ed.png)

Same on Seamonkey (well, except that I'm using the German locale there)
![saveas-sm](https://f.cloud.github.com/assets/383665/751866/c6fb57fe-e51d-11e2-8fc7-44167db21d6d.png)

Anyway, I consider this issue workforme and finished now.

**Comment by NessaishereDuck** [comment]
I agree Nils and thank you very much!! <3
